src/desktop/apps/article/components/InfiniteScrollNewsArticle.tsx

```diff
@@ -176,15 +172,28 @@ export class InfiniteScrollNewsArticle extends Component<
       if (hasNewDate) {
         this.setState({ date: nextArticle.published_at })
       }
-      document.title = 'News' // todo: replace with actual meta-title
-      window.history.replaceState({}, 'news', `/news`)
+      if (nextArticle.isTruncated === false) {
+        this.setMetadata(nextArticle)
+      } else {
+        this.setMetadata()
+      }
     }
   }
 
-  onExpand = (article) => {
-    // Set slug/document title when expanding an article
-    document.title = article.thumbnail_title
-    window.history.replaceState({}, article.id, `/news/${article.slug}`)
+  setMetadata = (article: any = null) => {
+    const id = article ? article.id : 'news'
+    const path = article ? `/news/${article.slug}` : '/news'
+    document.title = article ? article.thumbnail_title : 'News'
+    window.history.replaceState({}, id, path)
+  }
+
+  onExpand = (i) => {
+    const { articles } = this.state
+    const newArticles = cloneDeep(articles)
+    newArticles[i].isTruncated = false
+    this.setState({ articles: newArticles })
```

Comment:
Curious why you decided to set isTruncated as a property on the article object?  This is a pattern I haven't seen so much.  I wonder if it would make sense to break out the article chunk into a subcomponent so that we can store the truncated info in state.

Reply:
Ooh, that's interesting! I like that, or keeping a separate array of expanded articles. Tho state in the component seems to make more sense to me. Will continue hacking on this one.